torch/csrc/jit/export.cpp

```diff
@@ -235,9 +245,33 @@ void encodeGraph(onnx::GraphProto * p_g, const std::shared_ptr<Graph> & g,
     for(auto attr_name : node->attributeNames()) {
       addAttribute(p_n, node, attr_name);
     }
+    if (node->kind() == torch::jit::onnx::Loop) {
+      JIT_ASSERT(node->blocks().size() == 1);
+
+      auto body = p_n->add_attribute();
+      body->set_name("body");
+      body->set_type(onnx::aGRAPH);
+      auto g = body->mutable_g();
+      encodeBlock(g, node->blocks()[0], initializers, raw_data_export_map);
+    }
+    if (node->kind() == torch::jit::onnx::If) {
```

Comment:
It is a bit crappy that we need special cases in the protobuf translation. I can see why though -- we have numbered blocks in IR and named graphs in ONNX. I wonder if we can add "block attributes" to the IR (e.g. a `graph.b` function). They would simply hold the _index_ of the block, not the block itself, but would then translate into onnx automatically:

```
auto then_branch = graph->addBlock();
graph.b_(onnx::then_branch, 0);
auto else_branch = graph->addBlock();
graph.b_(onnx::then_branch, 1);
```

Then add a `::b` case to addAttribute.